### PR TITLE
Add package size

### LINKED ISSUE #359: Show install size of each package
I think it would be nice to show how large each package is before the user installs it.

For npm packages, we can get a badge from https://github.com/styfle/packagephobia/issues/24

For example:

```
<div class="description"><%= project.description %></div>
<div class="install-size">
 <img alt="install size" src="https://packagephobia.now.sh/badge?p=<%= project.name %>" />
</div>
```

Is `project.name` the correct property to find the name on npm?

### PR DESCRIPTION
**Pull request by styfle**
Fixes #359

### CODE REVIEW

File: scripts/fetch-archive.js [hunk 1]
```diff
@@ -39,6 +39,25 @@ async function getProjectGitHubData(repo) {
...
+    
+      res.on('end', () => {
+        resolve(JSON.parse(str));
+      });
+    
+    }).on('error', (err) => {
+      reject(error);
+    });
+  });
+}
```
Comment (erquhart) [inline scripts/fetch-archive.js]: We could shrink this a lot by using fetch.

File: scripts/fetch-archive.js [hunk 2]
```diff
@@ -49,16 +68,30 @@ async function getAllProjectGitHubData(repos) {
...
   const twitterFollowers = twitterScreenNames.length && await getTwitterFollowers(twitterScreenNames)
   const gitHubRepos = map(projects, 'repo').filter(val => val)
   const gitHubReposData = await getAllProjectGitHubData(gitHubRepos)
-  const data = projects.reduce((obj, { key, repo, twitter }) => {
-    const twitterData = twitter ? { followers: twitterFollowers[twitter] } : {}
+  const npms = map(projects, 'npm').filter(val => val);
+  const npmPackageData = await getAllProjectPackageSizes(npms);
+  
+  const data = projects.reduce((obj, { key, repo, npm, twitter }) => {
+    const twitterData = twitter ? { followers: twitterFollowers[twitter] }
```
Comment (erquhart) [inline scripts/fetch-archive.js]: Missing the `or` case for this ternary.

File: scripts/fetch-archive.js (hunk 1 above)
Comment (styfle) [reply scripts/fetch-archive.js]: Do you already have this as a dependency or do I need to add `node-fetch`?

**Comment by erquhart** [comment]
I actually wouldn't put this metric in the stats, maybe a "Package size" item in the list below?

**Comment by styfle** [comment]
That was my first thought too but I don't know how to put it in the list below 😅 

I just see the `fields` array but I'm not sure how to add a field in there 🤔

**Comment by erquhart** [comment]
It's actually pulled straight from the individual project frontmatters:

https://github.com/netlify/staticgen/blob/acf13dcf25b7769befda29b89e90c770ea5eafb4/static.config.js#L19

https://github.com/netlify/staticgen/blob/acf13dcf25b7769befda29b89e90c770ea5eafb4/static.config.js#L155-L161

It doesn't have to be just frontmatter though, you can append your data to the fields array in `src/Home/Project.js`.

**Comment by styfle** [comment]
Ok I see where `fields` come from: the site.yml file.
But where do the `fieldValues` come from?
And where should I add this new value?

**Comment by erquhart** [comment]
Right here: https://github.com/netlify/staticgen/blob/acf13dcf25b7769befda29b89e90c770ea5eafb4/static.config.js#L103

I'd say create a new `getFields` function that combines `siteConfig.fields` with additional fields like yours, then call that function in place of `siteConfig.fields` throughout.

To get your values into `fieldValues`, in that same `getProjects` function, you can map additional values onto `projectDetails` after this line:

https://github.com/netlify/staticgen/blob/acf13dcf25b7769befda29b89e90c770ea5eafb4/static.config.js#L88

**Comment by styfle** [comment]
Ok I tried a couple things but I'm still stuck 😭 
If you can make that change on this branch, feel free to do so 👍 
I won't have time to make another attempt for a week or so.

**Comment by erquhart** [comment]
Hey @styfle, just now circling back to this. I don't have much bandwidth to dig in - do you still plan on working on this?

**Comment by styfle** [comment]
Oops, I almost forgot about it 😅 
I will try again this weekend and see if I can conquer it ⛰

**Comment by erquhart** [comment]
@styfle closing this, but feel free to reopen if you ever circle back!

**Pull request closed without merging**